src/swap/saga.ts

```diff
@@ -118,3 +135,24 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
 export function* swapSaga() {
   yield takeLatest(swapStart.type, safely(swapSubmitSaga))
 }
+
+export function* sendApproveTx(tokenAddress: string, amount: string, recipientAddress: string) {
+  const kit: ContractKit = yield call(getContractKit)
+  const contract: Contract = yield call(getERC20TokenContract, tokenAddress)
+  const walletAddress: string = yield call(getConnectedUnlockedAccount)
+
+  const tx: CeloTransactionObject<boolean> = toTransactionObject(
+    kit.connection,
+    contract.methods.approve(recipientAddress, amount)
+  )
+  const preferredFeeCurrency: string | undefined = yield call(fetchFeeCurrencySaga)
```

Comment:
Oh, yeah, you are right!

Reply:
sorry i'm extremely late to the party but was it safe to remove this? i had added this because  `sendTransaction ` does not fill in the correct gas currency https://github.com/valora-inc/wallet/pull/3491 but i see here we haven't made any changes to the fee currency logic and just removed it?